tests/rgen/Microsoft.Macios.Generator.Tests/BaseGeneratorTestClass.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Xamarin.Tests;
+using Xamarin.Utils;
+using Xunit;
+
+namespace Microsoft.Macios.Generator.Tests;
+
+/// <summary>
+/// Base class that allows to test the generator.
+/// </summary>
+public class BaseGeneratorTestClass {
+	protected BindingSourceGeneratorGenerator GeneratorGenerator;
+	protected CSharpGeneratorDriver _driver;
+
+	readonly PortableExecutableReference [] _defaultReferences;
+
+	public BaseGeneratorTestClass ()
+	{
+		GeneratorGenerator = new BindingSourceGeneratorGenerator ();
+		_driver = CSharpGeneratorDriver.Create (GeneratorGenerator);
+
+		var dotNetAssemblyPath = Path.GetDirectoryName (typeof (object).Assembly.Location)!;
+		_defaultReferences =
+		[
+			MetadataReference.CreateFromFile (typeof (object).Assembly.Location),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "mscorlib.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Core.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Drawing.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Drawing.Primitives.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Runtime.InteropServices.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Private.CoreLib.dll")),
+			MetadataReference.CreateFromFile (Path.Combine (dotNetAssemblyPath, "System.Runtime.dll")),
+		];
```

Comment:
I think it would be better to use the assemblies in the `Configuration.DotNetBclDir` directory instead. In particular this isn't compatible with NativeAOT.